x-pack/plugin/mapper-version/build.gradle

```diff
@@ -4,12 +4,12 @@ apply plugin: 'elasticsearch.esplugin'
 apply plugin: 'elasticsearch.internal-cluster-test'
 
 esplugin {
-  name 'versionfield'
+  name 'version'
```

Comment:
I did not notice that the existing constant-keyword and flattened don't use the mapper prefix in here. I wonder why, maybe @jtibshirani has an idea? I was thinking that at least for version, it may make sense to use mapper-version here too and in the archivesBaseName, as version is very generic. Wondering what others think.

Reply:
That was my feeling as well. 'version' alone maybe sounds too generic.

Reply:
Recently we've omitted the 'mapper' prefix in plugin names, some examples: `constant-keyword`, `flattened`, `vectors`, `wildcard`.  I think we just changed the precedent, perhaps without clear intention to do so.

I also think using `mapper-version` at least in this one case to avoid confusion/ potential naming conflicts.